app/vmalert/main.go

```diff
@@ -358,11 +359,11 @@ func configReload(ctx context.Context, m *manager, groupsCfg []config.Group, sig
 		}
 		if configsEqual(newGroupsCfg, groupsCfg) {
 			templates.Reload()
-			// set success to 1 since previous reload
-			// could have been unsuccessful
+			// set success to 1 since previous reload could have been unsuccessful
+			// do not update configTimestamp as config version remains old.
 			configSuccess.Set(1)
-			setConfigError(nil)
-			// config didn't change - skip it
+			setLastConfigErr(nil)
```

Comment:
I think, some comment is needed here. Something like: 

 `reset an error in case when config changes was rolled back`.

 Because at first glance, it looks like it's an impossible case - if config are equal, code below should be called and it already resets an error.